src/store/app.module.js

```diff
@@ -27,6 +30,16 @@ const mutations = {
   setTitle (state, title) {
     state.title = title
   },
+  saveLayout (state, { workflowName, layout, views }) {
+    /* NOTE: use markRaw to prevent proxying of the Lumino layout in particular.
+    It is not necessary for this saved state to be reactive, and moreover
+    proxying the layout breaks some parts of the 3rd party Lumino backend. */
+    state.workspaceLayouts.set(workflowName, markRaw({ layout, views }))
+    if (state.workspaceLayouts.size > 10) {
```

Comment:
Would be worth building a really complex layout and finding out how heavy it is.

I'm hoping that we can safely raise this number.